interfaces/compatibility/compatibility_test.go

```diff
@@ -48,87 +48,64 @@ func (s *CompatSuite) TearDownTest(c *C) {
 }
 
 func (s *CompatSuite) TestValidCompatFields(c *C) {
-	for i, tc := range []struct {
-		compat string
-		cfield *compatibility.CompatField
-	}{
-		{"foo", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
-			{Tag: "foo", Values: []compatibility.CompatRange{{0, 0}}}}}},
-		{"foo-0", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
-			{Tag: "foo", Values: []compatibility.CompatRange{{0, 0}}}}}},
-		{"foo-4", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
-			{Tag: "foo", Values: []compatibility.CompatRange{{4, 4}}}}}},
-		{"foo-4-(1..3)", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
-			{Tag: "foo", Values: []compatibility.CompatRange{{4, 4}, {1, 3}}}}}},
-		{"foo-(1..3)-4", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
-			{Tag: "foo", Values: []compatibility.CompatRange{{1, 3}, {4, 4}}}}}},
-		{"foo3e4-4-(1..3)-bar-2025-8-2", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
-			{Tag: "foo3e4", Values: []compatibility.CompatRange{{4, 4}, {1, 3}}},
-			{Tag: "bar", Values: []compatibility.CompatRange{{2025, 2025}, {8, 8}, {2, 2}}},
-		}}},
-		{"foo3e4-4-(1..3)-bar-2025-8-2-libxxlongname-56-(4..90)", &compatibility.CompatField{Dimensions: []compatibility.CompatDimension{
-			{Tag: "foo3e4", Values: []compatibility.CompatRange{{4, 4}, {1, 3}}},
-			{Tag: "bar", Values: []compatibility.CompatRange{{2025, 2025}, {8, 8}, {2, 2}}},
-			{Tag: "libxxlongname", Values: []compatibility.CompatRange{{56, 56}, {4, 90}}},
-		}}},
+	for i, tc := range []string{
+		"foo",
+		"foo-0",
+		"foo-4",
+		"foo-4-(1..3)",
+		"foo-(1..3)-4",
+		"foo3e4-4-(1..3)-bar-2025-8-2",
+		"foo3e4-4-(1..3)-bar-2025-8-2-libxxlongname-56-(4..90)",
+		"(foo-3-4 OR bar-1-1-baz) AND boo32",
```

Comment:
we have parsing tests for this kind of expressions but I don't see them in the Expression logic test

Reply:
I've added a couple of tests to `TestCompatibilityExpressions`, but there were already some tests involving complex expressions, let me know if that is what you wanted.